docs/user_guide/building_blocks/selectors.md

```diff
@@ -0,0 +1,648 @@
+
+    # create the list of indices to be kept from the Jet collection using the jet_mask to create the
+    # new Jet field containing only the selected Jet objects
+    jet_indices = ak.local_index(events.Jet.pt)[jet_mask]
+
+    # create the list of indices to be kept from the Jet collection using the jet_50pt_mask to create the
+    # new Jet_50pt field containing only the selected Jet_50pt objects
+    jet_50pt_indices = ak.local_index(events.Jet.pt)[jet_50pt_mask]
+
+    return events, SelectionResult(
+        steps={
+            # boolean mask to create selection of the events with at least two jets, this will be
+            # applied in the ReduceEvents task
+            "jet": jet_sel,
+        },
+        objects={
+            # in ReduceEvents, the Jet field will be replaced by the new Jet field containing only
+            # selected jets, and a new field called Jet_50pt containing the jets with pt higher than
+            # 50 GeV will be created
+            "Jet": {
+                "Jet": jet_indices,
+                "Jet_50pt": jet_50pt_indices,
+            },
+        },
+        aux={
+            # jet mask that lead to the jet_indices
+            "jet_mask": jet_mask,
+        },
+    )
+
+
+# Next, define an internal fatjet Selector to be used by the exposed Selector
+
+@selector(
+    # define some additional information here, e.g.
+    # what columns are needed for this Selector?
+    uses={
+        "FatJet.pt",
+    },
+    # does this Selector produce any columns?
+    produces=set(),
+
+    # ...
+)
+def fatjet_selection_with_result(self: Selector, events: ak.Array, **kwargs) -> tuple[ak.Array, SelectionResult]:
+    # require an object of the FatJet collection to have at least 40 GeV pt to be
+    # considered a FatJet in our analysis
+    fatjet_mask = (events.FatJet.pt > 40.0)
+
+    # require an event to have at least one AK8-jet (=FatJet) to be selected
+    fatjet_sel = (ak.sum(fatjet_mask, axis=1) >= 1)
+
+    # create the list of indices to be kept from the FatJet collection using the fatjet_mask to create the
+    # new FatJet field containing only the selected FatJet objects
+    fatjet_indices = ak.local_index(events.FatJet.pt)[fatjet_mask]
+
+    return events, SelectionResult(
+        steps={
+            # boolean mask to create selection of the events with at least two jets, this will be
+            # applied in the ReduceEvents task
+            "fatjet": fatjet_sel,
+        },
+        objects={
+            # in ReduceEvents, the FatJet field will be replaced by the new FatJet field containing only
+            # selected fatjets
+            "FatJet": {
+                "FatJet": fatjet_indices,
+            },
+        },
+    )
+
+
+# Implement the task to update the stats object
+
+@selector(uses={"process_id", "mc_weight"})
+def increment_stats(
+    self: Selector,
+    events: ak.Array,
+    results: SelectionResult,
+    stats: dict,
+    **kwargs,
+) -> ak.Array:
```

Comment:
As this is a tutorial and there is a high probability that more weights or whatever additional information will be needed for the users in their own analysis I thought it would be better to present the mechanism here, I could still mention the already existing columnflow class if you wish... Do you find this section unnecessary?

Reply:
You make a good point that it is nice to show users how the stats are actually used instead of just explaining some magical columnflow function that does the work for you. In that way, they also see that they can just fill this dict with whatever they need themselved.
We can maybe discuss this on friday, but I have no clear preferences. But if we do not use the columnflow increment_stats, a link would probably be helpful.

Reply:
done (now, both approaches are presented)

Reply:
as both approaches are presented before, I have let it like that for now (It doesn't look that much easier with the columnflow selector in my opinion as you still need to create the weight map and the group map before... So presenting a more general approach with direct access to the object for the complete example seems more meaningful to me.)